### ISSUE TITLE
add option use system-wide tinyxml2

### ISSUE THREAD

**Issue by sl1pkn07** [KhronosGroup/Vulkan-Hpp#297]
Hi

is possible add the option for use system tinyxml2 instead of download the sources?

tinyxml2 have pkgconfig and cmake modules for use in cmake

greetings

**Comment by sl1pkn07** [comment]
~~~patch
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8f11117..05bfa7b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,19 +52,27 @@ set(SOURCES
   VulkanHppGenerator.cpp
 )
 
-set(TINYXML2_SOURCES
-  tinyxml2/tinyxml2.cpp
-)
-
-set(TINYXML2_HEADERS
-  tinyxml2/tinyxml2.h
-)
+find_package(tinyxml2 REQUIRED)
+if(tinyxml2_FOUND)
+  set(TINYXML2_HEADERS
+    ${tinyxml2_INCLUDE_DIRS}
+  )
+else()
+  set(TINYXML2_HEADERS
+    tinyxml2/tinyxml2.h
+  )
+  set(TINYXML2_SOURCES
+      tinyxml2/tinyxml2.cpp
+  )
+endif()
 
 source_group(headers FILES ${HEADERS})
 source_group(sources FILES ${SOURCES})
 
 source_group(TinyXML2\\headers FILES ${TINYXML2_HEADERS})
-source_group(TinyXML2\\sources FILES ${TINYXML2_SOURCES})
+if(NOT tinyxml2_FOUND)
+  set(TINYXML2_SOURCES
+      tinyxml2/tinyxml2.cpp
+  )
+endif()
 
 source_group(headers FILES ${HEADERS})
 source_group(sources FILES ${SOURCES})
 
 source_group(TinyXML2\\headers FILES ${TINYXML2_HEADERS})
-source_group(TinyXML2\\sources FILES ${TINYXML2_SOURCES})
+if(NOT tinyxml2_FOUND)
+  source_group(TinyXML2\\sources FILES ${TINYXML2_SOURCES})
+endif()
 
 add_executable(VulkanHppGenerator
   ${HEADERS}
@@ -75,7 +83,11 @@ add_executable(VulkanHppGenerator
 
 set_property(TARGET VulkanHppGenerator PROPERTY CXX_STANDARD 11)
 
-target_include_directories(VulkanHppGenerator PRIVATE "${CMAKE_SOURCE_DIR}/tinyxml2")
+if(tinyxml2_FOUND)
+  target_link_libraries(VulkanHppGenerator tinyxml2)
+elseif()
+  target_include_directories(VulkanHppGenerator PRIVATE "${CMAKE_SOURCE_DIR}/tinyxml2")
+endif()
 
 option (SAMPLES_BUILD OFF)
 if (SAMPLES_BUILD)
~~~

this would be ok?

**Comment by Wunkolo** [comment]
Just made a PR(#299) implementing a similar general implementation of what you posted that should try to find tinyxml2 on your system before resorting to the submodule.

**Comment by asuessenbach** [comment]
Resolved by #417.
